R/version_check.R

```diff
@@ -0,0 +1,50 @@
+# Utilities for detecting new version
+#
+# Author: kimyen
+###############################################################################
+
+.RAN <- "https://sage-bionetworks.github.io/ran"
+.PACKAGE_NAME = "synapser"
+
+# convert the full version to a precision
+# precision can be one of the following values:
+.MAJOR <- 1
+.MINOR <- 2
+.PATCH <- 3
+.simplifyVersion <- function(version, precision) {
+  if (!match(precision, c(.MAJOR, .MINOR, .PATCH))) {
+    throw("Invalid precision: ", precision)
+  }
+  parts <- unlist(strsplit(version, "[.]"))
+  parts[is.na(parts[1:precision])] <- "0"
+  return(paste(parts[1:precision], collapse = "."))
+}
+
+.printVersionOutOfDateWarnings <- function(current_version, latest_version) {
+  message <- sprintf("\nNew synapser version detected:
+  You are using synapser version %s.
+  synapser version %s is detected at %s.
+  To upgrade to the latest version of synapser, please run the following command:
+  install.packages(\"synapser\", repos=\"https://sage-bionetworks.github.io/ran\")\n",
+                     current_version, latest_version, .RAN)
+  packageStartupMessage(message)
+}
+
+# check package_name's version against old.packages() output to see if the
+# package version is out of date upto the given precision
+.isVersionOutOfDate <- function(info, package_name, precision) {
+  if (!is(info, "matrix") || !(package_name %in% rownames(info))) {
+    return(FALSE)
+  }
+  current_version <- .simplifyVersion(info[package_name, "Installed"], precision)
+  latest_version <- .simplifyVersion(info[package_name, "ReposVer"], precision)
+  return(current_version < latest_version)
```

Comment:
I don't think this line will correctly handle all cases, e.g.: 

``` r
current_version <- "0.5.0"
latest_version <- "0.10.0"
current_version < latest_version
#> [1] FALSE
```

It might be worth using R's built in `package_version` class:

```
package_version(current_version) < package_version(latest_version)
#> [1] TRUE
```

<sup>Created on 2019-02-26 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

Reply:
I will add that as a test case to my `testthat` cases. 
Also, I was looking at `compareVersion()`: https://stat.ethz.ch/R-manual/R-devel/library/utils/html/compareVersion.html